ethcore/src/miner/miner.rs

```diff
@@ -482,6 +489,7 @@ impl Miner {
 				// Save proposal for later seal submission and broadcast it.
 				Seal::Proposal(seal) => {
 					trace!(target: "miner", "Received a Proposal seal.");
+					*self.next_mandatory_reseal.write() = Instant::now() + self.options.reseal_max_period;
```

Comment:
Sealing may sometimes fail when the `Engine` does not generate a seal, so it would be good to keep trying.